src/frontend/src/handler/create_source.rs

```diff
@@ -696,6 +666,76 @@ pub(crate) async fn bind_columns_from_source(
     Ok(res)
 }
 
+/// add connector-spec columns to the end of column catalog
+pub fn handle_addition_columns(
+    with_properties: &HashMap<String, String>,
+    mut include_columns_options: Vec<(Ident, Option<Ident>)>,
+    columns: &mut Vec<ColumnCatalog>,
+) -> Result<()> {
+    // refer the logic from `parse_source_schema_with_connector`
+    const EXEMPTED_CONNECTORS: [&str; 4] = ["nexmark", "datagen", "google_pubsub", "test"];
+    let is_connector_exempted_from_additional_column = |connector_name: &str| -> bool {
+        connector_name.contains("-cdc") || EXEMPTED_CONNECTORS.contains(&connector_name)
+    };
```

Comment:
Why we need this instead of using `CONNECTOR_COMPATIBLE_ADDITIONAL_COLUMNS` to handle all it?